openmsupply/application-manager-server
# #416 Verification -- endpoint and action

Fixes #416 

Implements solution as suggested in [the issue](https://github.com/openmsupply/application-manager-server/issues/416), with some amendments basic on subsequent [discussion](https://github.com/openmsupply/application-manager-server/issues/416#issuecomment-866442834)

Requires [front-end PR #810](https://github.com/openmsupply/application-manager-web-app/pull/810)

**Note**: the additional fields on the "verification" table (code, data, etc.) exist, and can be passed as parameters, but I'm not doing anything with them yet.

### Testing

- make sure you've set up for the Send Email action to work as described in my previous PR #422 
- fresh `yarn database init dev`
- from Login page, click "create new account" and submit a new user registration form (I've put the more complex user reg into "laos" folder, and used the shortened one here in "dev")
- On submission, you should receive a verification email. Note that the user has not yet been created in the system (or any other "approval" actions)
- Click the verification link. You should see a "Successful verification" page, and the user has been created, so you can now log in with your new user account
